src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.VariableIdentifier.cs

```diff
@@ -16,9 +16,17 @@ internal partial class LocalDataFlowPass<TLocalState>
 
             public VariableIdentifier(Symbol symbol, int containingSlot = 0)
             {
-                Debug.Assert(symbol.Kind == SymbolKind.Local || symbol.Kind == SymbolKind.Field || symbol.Kind == SymbolKind.Parameter ||
-                    (symbol as MethodSymbol)?.MethodKind == MethodKind.LocalFunction ||
-                    symbol.Kind == SymbolKind.Property || symbol.Kind == SymbolKind.Event);
+                Debug.Assert(symbol.Kind switch
+                {
+                    SymbolKind.Local => true,
```

Comment:
Unrelated to the PR: this makes me really want `x switch { 1, 2, 3, 4, 5 => true }`

Reply:
In the works for C# 9, though the syntax would more likely be `1 or 2 or 3 or 4 or 5`... also (I hope) you could avoid the match altogether and write `x is >= 1 and <= 5`

Reply:
With relational patterns and pattern combinators that would be

`x is >= 1 and <= 5`

---
In reply to: [267972777](https://github.com/dotnet/roslyn/pull/34273#discussion_r267972777) [](ancestors = 267972777)